nms/app/packages/magmalte/app/views/subscriber/__tests__/SubscriberTest.js

```diff
@@ -0,0 +1,137 @@
+import {cleanup, fireEvent, render, wait} from '@testing-library/react';
+
+jest.mock('axios');
+jest.mock('@fbcnms/magma-api');
+jest.mock('@fbcnms/ui/hooks/useSnackbar');
+afterEach(cleanup);
+const enqueueSnackbarMock = jest.fn();
+jest
+  .spyOn(require('@fbcnms/ui/hooks/useSnackbar'), 'useEnqueueSnackbar')
+  .mockReturnValue(enqueueSnackbarMock);
+
+describe('<SubscriberDashboard />', () => {
+  const subscribers = {
+    IMSI00000000001002: {
+      active_apns: ['oai.ipv4'],
+      id: 'IMSI722070171001002',
+      lte: {
+        auth_algo: 'MILENAGE',
+        auth_key: 'i69HPy+P0JSHzMvXCXxoYg==',
+        auth_opc: 'jie2rw5pLnUPMmZ6OxRgXQ==',
+        state: 'ACTIVE',
+        sub_profile: 'default',
+      },
+      config: {
+        lte: {
+          auth_algo: 'MILENAGE',
+          auth_key: 'i69HPy+P0JSHzMvXCXxoYg==',
+          auth_opc: 'jie2rw5pLnUPMmZ6OxRgXQ==',
+          state: 'ACTIVE',
+          sub_profile: 'default',
+        },
+      },
+    },
+    IMSI00000000001003: {
+      active_apns: ['oai.ipv4'],
+      id: 'IMSI722070171001003',
+      lte: {
+        auth_algo: 'MILENAGE',
+        auth_key: 'i69HPy+P0JSHzMvXCXxoYg==',
+        auth_opc: 'jie2rw5pLnUPMmZ6OxRgXQ==',
+        state: 'INACTIVE',
+        sub_profile: 'default',
+      },
+      config: {
+        lte: {
+          auth_algo: 'MILENAGE',
+          auth_key: 'i69HPy+P0JSHzMvXCXxoYg==',
+          auth_opc: 'jie2rw5pLnUPMmZ6OxRgXQ==',
+          state: 'INACTIVE',
+          sub_profile: 'default',
+        },
+      },
+    },
+  };
+
+  const Wrapper = () => {
+    const subscriberCtx = {
+      state: subscribers,
+      gwSubscriberMap: {},
+    };
+
+    return (
+      <MemoryRouter
+        initialEntries={['/nms/test/subscribers/overview']}
+        initialIndex={0}>
+        <MuiThemeProvider theme={defaultTheme}>
+          <MuiStylesThemeProvider theme={defaultTheme}>
+            <SubscriberContext.Provider value={subscriberCtx}>
+              <Route
+                path="/nms/:networkId/subscribers/overview"
+                component={SubscriberDashboard}
+              />
+            </SubscriberContext.Provider>
+          </MuiStylesThemeProvider>
+        </MuiThemeProvider>
+      </MemoryRouter>
+    );
+  };
+
+  it('Verify Subscribers Dashboard', async () => {
+    const {getAllByTitle, getByTestId, getAllByRole} = render(<Wrapper />);
+    await wait();
+    const rowItems = await getAllByRole('row');
+
+    // first row is the header
+    expect(rowItems[0]).toHaveTextContent('Name');
+    expect(rowItems[0]).toHaveTextContent('IMSI');
+    expect(rowItems[0]).toHaveTextContent('Service');
+    expect(rowItems[0]).toHaveTextContent('Current Usage');
+    expect(rowItems[0]).toHaveTextContent('Daily Average');
+    expect(rowItems[0]).toHaveTextContent('Last Reported Time');
+
+    expect(rowItems[1]).toHaveTextContent('IMSI00000000001002');
+    expect(rowItems[1]).toHaveTextContent('IMSI00000000001002');
```

Comment:
Does the repeated expectation verify `Name` and `IMSI` separately? I'm just looking off of `https://github.com/testing-library/jest-dom#tohavetextcontent` and perhaps it's something undocumented?

Reply:
No.. it's doing the same thing twice